pkg/oss/nodeserver.go

```diff
@@ -367,9 +369,25 @@ func (ns *nodeServer) NodeUnpublishVolume(ctx context.Context, req *csi.NodeUnpu
 	if err != nil {
 		return nil, err
 	}
-	if !IsOssfsMounted(mountPoint) {
-		log.Infof("Directory is not mounted: %s", mountPoint)
-		return &csi.NodeUnpublishVolumeResponse{}, nil
+
+	// check mount point with IsLikelyNotMountPoint first
+	notmounted, err := ns.k8smounter.IsLikelyNotMountPoint(mountPoint)
+	if err != nil {
+		log.Errorf("NodeUnpublishVolume:: use IsLikelyNotMountPoint to check if path %s mounted failed with error %v", mountPoint, err)
+		notmounted = !IsOssfsMounted(mountPoint)
+	}
+
+	if notmounted {
+		if empty, _ := IsDirEmpty(mountPoint); empty {
+			if removeErr := os.Remove(mountPoint); removeErr != nil {
```

Comment:
How about just call `os.Remove`? You don't need to check mounted or empty first, since `os.Remove` would fail in those cases.